build/Build.Steps.cs

```diff
@@ -66,15 +66,43 @@ partial class Build
         .Unlisted()
         .Executes(() => ControlFlow.ExecuteWithRetry(() =>
         {
+            var projectsToRestore = Solution.GetCrossPlatformManagedProjects();
+
             if (IsWin)
             {
-                DotNetRestore(s => s
-                    .SetProjectFile(Solution)
-                    .SetVerbosity(DotNetVerbosity.Normal)
-                    .SetProperty("configuration", BuildConfiguration.ToString())
-                    .When(!string.IsNullOrEmpty(NugetPackageDirectory), o =>
-                        o.SetPackageDirectory(NugetPackageDirectory)));
+                projectsToRestore = projectsToRestore.Concat(Solution.GetWindowsOnlyTestApplications());
+            }
 
+            foreach (var project in projectsToRestore)
+            {
+                if (TestPackageVersions.Versions.TryGetValue(project.Name, out var testedPackageVersions))
+                {
+                    foreach (var testedPackageVersion in testedPackageVersions)
+                    {
+                        DotNetRestore(s => s
+                            .SetProjectFile(project)
+                            .SetVerbosity(DotNetVerbosity.Normal)
+                            .SetProperty("configuration", BuildConfiguration.ToString())
+                            .SetProperty("TestedPackageVersion", testedPackageVersion)
```

Comment:
Just calling attention that this means that only a single package can "rotate" versions on the test app. It is reasonable, if there is a need to test different packages in a single app it is not hard to adapt the code to do that.